src/components/ble/CurrentTimeService.cpp

```diff
@@ -18,47 +30,92 @@ void CurrentTimeService::Init() {
   ASSERT(res == 0);
 
   res = ble_gatts_add_svcs(serviceDefinition);
+
   ASSERT(res == 0);
 }
 
-int CurrentTimeService::OnTimeAccessed(uint16_t conn_handle, uint16_t attr_handle, struct ble_gatt_access_ctxt* ctxt) {
+int CurrentTimeService::OnCurrentTimeAccessed(uint16_t conn_handle, uint16_t attr_handle, struct ble_gatt_access_ctxt* ctxt) {
 
   NRF_LOG_INFO("Setting time...");
 
   if (ctxt->op == BLE_GATT_ACCESS_OP_WRITE_CHR) {
-    CtsData result;
-    os_mbuf_copydata(ctxt->om, 0, sizeof(CtsData), &result);
+    CtsCurrentTimeData result;
+    int res = os_mbuf_copydata(ctxt->om, 0, sizeof(CtsCurrentTimeData), &result);
+
+    if (res < 0) {
+      NRF_LOG_ERROR("Error reading BLE Data writing to CTS Current Time (too little data)")
+      // return OK/0 anyway, we did consume the request, we just ignored the invalid value´
```

Comment:
I think invalid packets should return an error code to the client.

Reply:
I guess I was mixed up in different BLE implementations and thought that returning !=0 means request was not handled.
Fixed.